### PR TITLE
React to `AckSubtaskResultsVerify` and upload resources + results

### LINKED ISSUE #2190: [Concent[ UC4 / Additional Verification - react to `AckSubtaskResultsVerify`
as a Provider, implement reaction to the Concent's `AckSubtaskResultsVerify`

### LINKED ISSUE #2191: [Concent] UC4 / Additional Verification - as a Provider, upload files to the Concent
implement uploading files to the Concent service for the purpose of additional verifiaction, using the `FileTransferToken` obtained in the `AckSubtaskResultsVerify` response.

### PR DESCRIPTION
**Pull request by shadeofblue**
- `helpers.verify_message_signature` -> uses `Message.verify_signature` instead
* + `AckSubtaskResultsVerify` handler
* received_handler DRY
* concent filestransfers: recognize file category
* concent message handlers: upload resources vs upload results
* add task_id -> resource package resolver to CompTaskKeeper's mappings

closes #2190 
closes #2191

### CODE REVIEW

File: golem/network/concent/received_handler.py [hunk 1]
```diff
@@ -121,6 +121,29 @@ def concent_service(self):
     def concent_filetransfers(self) -> ConcentFiletransferService:
         return self.task_server.client.concent_filetransfers
 
+    def is_ours(self,
+                parent_msg: message.base.Message,
+                child_msg_field: str):
```
Comment (etam) [inline golem/network/concent/received_handler.py]: return type annotation?

File: golem/network/concent/received_handler.py [hunk 2]
```diff
@@ -320,6 +343,61 @@ def on_ack_force_get_task_result(self, msg, **_):
     def _log_ftt_invalid(msg: message.base.Message):
         logger.warning("File Transfer Token invalid in %r", msg)
 
+    def _upload_results(self, subtask_id, ftt):
```
Comment (etam) [inline golem/network/concent/received_handler.py]: type annotations?

File: golem/network/concent/received_handler.py [hunk 3]
```diff
@@ -320,6 +343,61 @@ def on_ack_force_get_task_result(self, msg, **_):
...
+        self.concent_filetransfers.transfer(
+            file_path=wtr.result_path,
+            file_transfer_token=ftt,
+            success=success,
+            error=error,
+            file_category=message.concents.FileTransferToken.
+            FileInfo.Category.results
+        )
+
+    def _upload_task_resources(self, task_id, ftt):
```
Comment (etam) [inline golem/network/concent/received_handler.py]: type annotations?

File: golem/task/taskkeeper.py [hunk 4]
```diff
@@ -267,12 +280,24 @@ def remove_old_tasks(self):
...
                 del self.subtask_to_task[subtask_id]
+
             del self.active_tasks[task_id]
 
+            if task_id in self.task_package_paths:
+                del self.task_package_paths[task_id]
+
         self.dump()
 
+    def add_package_paths(self, task_id, package_paths):
```
Comment (etam) [inline golem/task/taskkeeper.py]: type annotations?

File: golem/task/taskkeeper.py [hunk 5]
```diff
@@ -267,12 +280,24 @@ def remove_old_tasks(self):
...
+        self.task_package_paths[task_id] = package_paths
+        self.dump()
+
+    def get_package_paths(self, task_id):
```
Comment (etam) [inline golem/task/taskkeeper.py]: type annotations?

File: tests/golem/network/concent/test_filetransfers.py [hunk 6]
```diff
@@ -164,11 +180,15 @@ def test_process_error_no_handler(self):
...
+    def _init_uploaded_file(path):
+        with open(path, 'w') as f:
+            f.write('meh')
+
     @mock.patch('golem.network.concent.filetransfers.requests.post')
     def test_upload(self, requests_mock):
         path = self.path + '/something.good'
-        with open(path, 'w') as f:
-            f.write('meh')
+        self._init_uploaded_file(path)
```
Comment (etam) [inline tests/golem/network/concent/test_filetransfers.py]: You can use `self.new_path`, which is a `pathlib.Path`. Then everything is easy:

```python
(self.new_path / "something.good").write_text('meh')
```

**Review by jiivan — APPROVED** [review]

File: tests/golem/network/concent/test_filetransfers.py [hunk 7]
```diff
@@ -191,6 +211,35 @@ def test_upload(self, requests_mock):
         self.assertIsNotNone(kwargs.get('headers').pop('Concent-Auth'))
         self.assertEqual(kwargs.get('headers'), headers)
 
+    @mock.patch('golem.network.concent.filetransfers.requests.post')
+    def test_upload_multiple_files(self, requests_mock):
+        path = self.path + '/obsta.cles'
+        self._init_uploaded_file(path)
```
Comment (etam) [inline tests/golem/network/concent/test_filetransfers.py]: `self.new_path` like above

**Review by etam — CHANGES_REQUESTED** [review]

File: tests/golem/network/concent/test_filetransfers.py [hunk 8]
```diff
@@ -164,11 +180,14 @@ def test_process_error_no_handler(self):
         with self.assertRaises(Exception):
             self.cfs.process(request)
 
+    def _init_uploaded_file(self, filename: str):
```
Comment (etam) [inline tests/golem/network/concent/test_filetransfers.py]: return type annotation plz

File: golem/task/taskkeeper.py [hunk 9]
```diff
@@ -267,12 +280,24 @@ def remove_old_tasks(self):
...
+            if task_id in self.task_package_paths:
+                del self.task_package_paths[task_id]
+
         self.dump()
 
+    def add_package_paths(self, task_id: str, package_paths: list) -> None:
+        self.task_package_paths[task_id] = package_paths
+        self.dump()
+
+    def get_package_paths(self, task_id: str) -> typing.Optional[list]:
```
Comment (etam) [inline golem/task/taskkeeper.py]: In type annotations don't use `list`. Use `typing.List[inner_type]`.

File: golem/task/taskkeeper.py [hunk 10]
```diff
@@ -267,12 +280,24 @@ def remove_old_tasks(self):
...
                 del self.subtask_to_task[subtask_id]
+
             del self.active_tasks[task_id]
 
+            if task_id in self.task_package_paths:
+                del self.task_package_paths[task_id]
+
         self.dump()
 
+    def add_package_paths(self, task_id: str, package_paths: list) -> None:
```
Comment (etam) [inline golem/task/taskkeeper.py]: In type annotations don't use `list`. Use `typing.List[inner_type]`.

**Review by etam — CHANGES_REQUESTED** [review]

File: tests/golem/network/concent/test_filetransfers.py (hunk 8 above)
Comment (shadeofblue) [reply tests/golem/network/concent/test_filetransfers.py]: @etam uhm this is a test, mind you...

**Review by etam — APPROVED** [review]
ok

File: tests/golem/network/concent/test_filetransfers.py (hunk 8 above)
Comment (etam) [reply tests/golem/network/concent/test_filetransfers.py]: No code is not worth a type annotation.

**Pull request merged**